content/posts/kubernetes-chargeback.md

```diff
@@ -0,0 +1,190 @@
+
+Storing this information in a suitable database allows us then, to create the desired reports, dashboards, alerts and chargebacks into our organization.
+
+## Proposed Toolchain
+
+There are some companies providing tools to enable transparency.
+
+You may have much of the required tools already available in your Kubernetes clusters. In fact, a powerful toolchain enabling powerful cost transparency queries and reports uses Prometheus at the core, which is often deployed with Kubernetes clusters for monitoring purposes already. Prometheus is a time-series database and allows efficient ingestion, storing and querying of time series data, such as resource usage data which is scraped on a fixed interval such as once a minute.
+
+The Prometheus ecosystem also contains the following tools which serve our use case well and are often installed with it.
+
+- The Prometheus Node Exporter measures CPU and memory usage of pods and containers as well as other metrics on nodes in a Kubernetes cluster.
+- Kube-State-Metrics prepares other Kubernetes information for ingestion by Prometheus.
+- The Prometheus Alertmanager allows us to alert based on specific thresholds and send emails, trigger a pager or generate a ticket.
+- Grafana enables us to create dashboards to visualize resource usage over time and interactive cost exploration.
+
+### Kubecost
+
+Kubecost is an open source cost monitoring tool for Kubernetes that is also based on Prometheus and it's ecosystem and can already give you insight per namespace.
+
+It runs on your cluster, offers basic reports and dashboards and makes some recommendations to rightsize specific workloads.
+
+## Introducing A Cost Management Solution In Your Organization
+
+While the proposed toolchain fulfills the technical needs to gain insight, I want to discuss the path for introducing a cost management solution in your organization.
+
+### Starting To Extract The Data
+
+As mentioned before, Kubernetes does not persist any of the data required for cost aggregation. The first step is therefore to put the infrastructure for tracking and storing this data in place. Your historical data will now start building and allow you insight on a weekly, monthly and eventually yearly basis.
+
+### Create First Reports
+
+Once resource usage data is available for a short time period, a week is a good start, you'll be able to create first reports and start identifying some of the bigger resources consumers, which you may approach to ensure their consumption is expedient for your business.
+
+### Workload Attribution
+
+At this point you may notice that many of the workloads do not clearly belong to a specific team, organizational unit, project or cost center. You'll have to find this information in your organization and then **tag** these resources in order to automatically attribute costs to the right cost center in the future.
```

Comment:
Shouldn't we use the term "label" instead of "tag" as it's the Kubernetes term for that?